example/src/main/java/org/folio/tlib/example/MyApi.java

```diff
@@ -98,21 +97,27 @@ private Future<Void> getTitles(Vertx vertx, RoutingContext ctx) {
     if (orderBy != null) {
       sql = sql + " ORDER BY " + orderBy;
     }
-    return pool.query(sql).execute().onSuccess(rows -> {
-      RowIterator<Row> iterator = rows.iterator();
-      JsonArray titles = new JsonArray();
-      while (iterator.hasNext()) {
-        Row row = iterator.next();
-        titles.add(new JsonObject()
-            .put("id", row.getUUID("id").toString())
-            .put("title", row.getString("title"))
-        );
-      }
-      ctx.response().putHeader("Content-Type", "application/json");
-      ctx.response().setStatusCode(200);
-      JsonObject result = new JsonObject().put("titles", titles);
-      ctx.response().end(result.encode());
-    }).mapEmpty();
+    return sql;
+  }
+
+  private Future<Void> getBooks(Vertx vertx, RoutingContext ctx) {
```

Comment:
@adamdickmeiss While this illustrates the use of object mapping and templates it doesn't really shows why a developer should bother with the extra boilerplate. 

Could we split this up into a a Service/Api and a Storage/Repo class and move the method to the storage class with a Future<List<Books>> signature to illustrate layering of code?